engine/dlib/src/test/test_httpclient.cpp

```diff
@@ -848,6 +848,53 @@ TEST_P(dmHttpClientTest, MaxAgeCache)
     ASSERT_EQ(dmHttpCache::RESULT_OK, cache_r);
 }
 
+TEST_P(dmHttpClientTest, TrustCache)
+{
+    dmHttpClient::Delete(m_Client);
+
+    // Reinit client with http-cache
+    dmHttpClient::NewParams params;
+    params.m_Userdata = this;
+    params.m_HttpContent = dmHttpClientTest::HttpContent;
+    params.m_HttpHeader = dmHttpClientTest::HttpHeader;
+    params.m_DNSChannel = m_DNSChannel;
+    params.m_TrustCache = true;
+    dmHttpCache::NewParams cache_params;
+    cache_params.m_Path = "tmp/cache";
+    dmHttpCache::Result cache_r = dmHttpCache::Open(&cache_params, &params.m_HttpCache);
+    ASSERT_EQ(dmHttpCache::RESULT_OK, cache_r);
+    m_Client = dmHttpClient::New(&params, m_URI.m_Hostname, m_URI.m_Port, strcmp(m_URI.m_Scheme, "https") == 0);
+    ASSERT_NE((void*) 0, m_Client);
+
+    dmHttpClient::Result r;
+    r = dmHttpClient::Get(m_Client, "/max-age-cached");
+    ASSERT_TRUE((r == dmHttpClient::RESULT_OK && m_StatusCode == 200) || (r == dmHttpClient::RESULT_NOT_200_OK && m_StatusCode == 304));
+    std::string val = m_Content;
+
+    // The web-server is returning System.currentTimeMillis()
+    // Ensure next ms
+    dmTime::Sleep(1U);
+
+    m_Content = "";
+    r = dmHttpClient::Get(m_Client, "/max-age-cached");
+    ASSERT_TRUE((r == dmHttpClient::RESULT_OK && m_StatusCode == 200) || (r == dmHttpClient::RESULT_NOT_200_OK && m_StatusCode == 304));
```

Comment:
How does this test that no actual http request was made?

Reply:
It doesn't really. We do however know that since we waited longer than max age the result should be different if a new request had been made to the server.

Reply:
Wasn't that what the other test did too?
What does this test do differently?

Reply:
They are quite similar and could probably be combined into a single test.

The new test checks that we get the same value back from the cache even if max-age time has elapsed (ie no round-trip to the server) when the `trust_cache` option is used.

The other test checks that we actually get a new value when max-age time has elapsed (ie we do a round-trip to the server).